roles/configure_snapraid/tasks/snapper_config.yml

```
@@ -38,61 +32,55 @@
+          ignore_errors: true
 
-    - name: Remove .snapshots subvolume for failed configs  # noqa command-instead-of-module
-      ansible.builtin.command:
-        cmd: btrfs subvolume delete -C {{ data_mount_path }}/data{{ '%02d' % item }}/.snapshots
-      loop: "{{ range(1, data_disks | length + 1) }}"
-      when:
-        - data_disks is defined
-        - data_disks|length > 0
-        - not config_validity.results[item-1].ansible_facts.config_valid
-        - "'creating btrfs subvolume .snapshots failed since it already exists' in create_config_result.results[item-1].stderr"
-      failed_when: false
+        - name: Attempt to create Snapper configuration
+          ansible.builtin.command:
+            cmd: snapper -c data{{ '%02d' % item }} create-config -t default {{ data_mount_path }}/data{{ '%02d' % item }}
+          register: create_config_result
+          changed_when: create_config_result.rc == 0
+          ignore_errors: true
+          loop: "{{ range(1, data_disks | length + 1) }}"
+          when: not config_validity.results[item-1].ansible_facts.config_valid
 
-    - name: Retry creating Snapper configuration for failed configs
-      ansible.builtin.command:
-        cmd: snapper -c data{{ '%02d' % item }} create-config -t default {{ data_mount_path }}/data{{ '%02d' % item }}
-      loop: "{{ range(1, data_disks | length + 1) }}"
-      when:
-        - data_disks is defined
-        - data_disks|length > 0
-        - not config_validity.results[item-1].ansible_facts.config_valid
-        - "'creating btrfs subvolume .snapshots failed since it already exists' in create_config_result.results[item-1].stderr"
-      failed_when: false
+        - name: Remove existing snapshots for failed configs  # noqa command-instead-of-module
+          ansible.builtin.command:
+            cmd: btrfs subvolume delete -C {{ data_mount_path }}/data{{ '%02d' % item }}/.snapshots/*
```

This is incorrect. My understanding is that each snapshot needs to be specified: `.snapshots/(snapshot number)/snapshot <-- this is the snapshot. 

```bash
❯ sudo btrfs subvolume delete -C /mnt/data-disks/data01/.snapshots/*
zsh: no matches found: /mnt/data-disks/data01/.snapshots/*
❯ sudo btrfs subvolume delete -C /mnt/data-disks/data01/.snapshots/1/snapshot
Delete subvolume 262 (commit): '/mnt/data-disks/data01/.snapshots/1/snapshot'
```

Having said this, I'm not sure the original worked in which case but it must have?

---

Good spot, I think this likely never worked and the failure was just masked by `failed_when: false.

Per your suggestion in DMs, I'll alter it to build a list of snaps and loop over that to delete them.

---

Good spot, I think this likely never worked and the failure was just masked by `failed_when: false.

Per your suggestion in DMs, I'll alter it to build a list of snaps and loop over that to delete them.

---

This should be resolved by babd86c45d71e723c5f301e53287f05846101c5e.